File: backend_node/src/controllers/tareas.controller.js
import { getConnection, sql } from "../config/db.js";

// OBTENER
export const obtenerTareas = async (req, res) => {
  try {
    const uid = req.user.uid;

    const pool = await getConnection();

    const result = await pool
      .request()
      .input("uid", sql.VarChar, uid)
      .query("SELECT * FROM tasks WHERE uid = @uid");

    res.json(result.recordset);
  } catch (error) {
    console.error("Error obteniendo tareas:", error);
    res.status(500).json({ error: "Error al obtener tareas" });
  }
};

// CREAR
export const crearTarea = async (req, res) => {
  try {
    const uid = req.user.uid;
    const { titulo, descripcion } = req.body;

    const pool = await getConnection();

    await pool
      .request()
      .input("uid", sql.VarChar, uid)
      .input("titulo", sql.VarChar, titulo)
      .input("descripcion", sql.VarChar, descripcion)
      .input("estado", sql.VarChar, "pendiente")
      .query(`
        INSERT INTO tasks (uid, titulo, descripcion, estado)
        VALUES (@uid, @titulo, @descripcion, @estado)
      `);

    res.json({ msg: "Tarea creada" });
  } catch (error) {
    console.error("Error creando tarea:", error);
    res.status(500).json({ error: "Error al crear tarea" });
  }
};

// ACTUALIZAR
export const actualizarTarea = async (req, res) => {
  try {
    const { id } = req.params;
    const { titulo, descripcion, estado } = req.body;

    const pool = await getConnection();

    await pool
      .request()
      .input("id", sql.Int, id)
      .input("titulo", sql.VarChar, titulo)
      .input("descripcion", sql.VarChar, descripcion)
      .input("estado", sql.VarChar, estado)
      .query(`
        UPDATE tasks 
        SET titulo=@titulo, descripcion=@descripcion, estado=@estado
        WHERE task_id=@id
      `);

    res.json({ msg: "Actualizada" });
  } catch (error) {
    console.error("Error actualizando tarea:", error);
    res.status(500).json({ error: "Error al actualizar" });
  }
};

// ELIMINAR
export const eliminarTarea = async (req, res) => {
  try {
    const { id } = req.params;

    const pool = await getConnection();

    await pool
      .request()
      .input("id", sql.Int, id)
      .query("DELETE FROM tasks WHERE task_id=@id");

    res.json({ msg: "Eliminada" });
  } catch (error) {
    console.error("Error eliminando tarea:", error);
    res.status(500).json({ error: "Error al eliminar" });
  }
};

File: backend_python/src/controllers/tareas_controller.py
from src.config.db import get_connection
from fastapi import HTTPException



# OBTENER
def obtener_tareas(uid):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("SELECT * FROM tasks WHERE uid = %s", (uid,))
    tareas = cursor.fetchall()

    conn.close()
    return tareas



# CREAR
def crear_tarea(uid, data):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("""
        INSERT INTO tasks (uid, titulo, descripcion, estado)
        VALUES (%s, %s, %s, %s)
    """, (uid, data.get("titulo"), data.get("descripcion"), "pendiente"))

    conn.commit()
    conn.close()

    return {"mensaje": "Tarea creada"}



# ACTUALIZAR
def actualizar_tarea(id, data):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("""
        UPDATE tasks
        SET titulo=%s, descripcion=%s, estado=%s
        WHERE task_id=%s
    """, (
        data.get("titulo"),
        data.get("descripcion"),
        data.get("estado"),
        id
    ))

    conn.commit()
    conn.close()

    return {"mensaje": "Actualizada"}


# ELIMINAR

def eliminar_tarea(id):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("DELETE FROM tasks WHERE task_id=%s", (id,))

    conn.commit()
    conn.close()

    return {"mensaje": "Eliminada"}